frappe/core/doctype/docfield/docfield.json

```diff
@@ -559,13 +560,19 @@
    "fieldname": "sort_options",
    "fieldtype": "Check",
    "label": "Sort Options"
+  },
+  {
+   "default": "0",
+   "fieldname": "not_nullable",
+   "fieldtype": "Check",
```

Comment:
Hide it in UI when it doesn't make sense:
- Fieldtypes that are of "no value" types
- Fieldtypes that always have defaults in framework - numeric types.